crates/index-scheduler/src/scheduler/process_snapshot_creation.rs

```diff
@@ -206,4 +230,339 @@ impl IndexScheduler {
+                    Err(_) => {
+                        return Err(Error::S3Error {
+                            status: resp.status(),
+                            body: resp.text().await.unwrap_or_default(),
+                        })
+                    }
+                };
+                let etag =
+                    resp.headers().get(ETAG).expect("every UploadPart request returns an Etag");
+                etags.push(bump.alloc_str(etag.to_str().unwrap()));
+            }
+
+            tracing::trace!("Finalizing the multipart upload");
+
+            let action = bucket.complete_multipart_upload(
+                Some(&credential),
+                &object,
+                multipart.upload_id(),
+                etags.iter().map(AsRef::as_ref),
+            );
+            let url = action.sign(s3_signature_duration);
+            let body = action.body();
+            let resp = backoff::future::retry(retry_backoff, move || {
+                let client = client.clone();
+                let url = url.clone();
+                let body = body.clone();
+                async move {
+                    match client.post(url).body(body).send().await {
+                        Ok(resp) if resp.status().is_client_error() => {
+                            resp.error_for_status().map_err(backoff::Error::Permanent)
+                        }
+                        Ok(resp) => Ok(resp),
+                        Err(e) => Err(backoff::Error::transient(e)),
+                    }
+                }
+            })
+            .await
+            .map_err(Error::S3HttpError)?;
+
+            let status = resp.status();
+            let body =
+                resp.text().await.map_err(|e| Error::S3Error { status, body: e.to_string() })?;
+
+            if status.is_success() {
+                Ok(())
+            } else {
+                Err(Error::S3Error { status, body })
+            }
+        });
+
+        let index_scheduler = IndexScheduler::private_clone(self);
+        let builder_task = tokio::task::spawn_blocking(move || {
+            let writer = flate2::write::GzEncoder::new(writer, flate2::Compression::new(level));
+            let mut tarball = tar::Builder::new(writer);
+
+            // 1. Snapshot the version file
+            tarball.append_path_with_name(
+                &index_scheduler.scheduler.version_file_path,
+                VERSION_FILE_NAME,
+            )?;
+
+            // 2. Snapshot the index scheduler LMDB env
+            progress.update_progress(SnapshotCreationProgress::SnapshotTheIndexScheduler);
+            let mut tasks_env_file = index_scheduler.env.try_clone_inner_file()?;
+            // Note: Seeking to the start of the file is necessary to ensure
+            //       the tarball reads the file from the beginning. I still
+            //       don't know why the read cursor is not at the beginning.
+            tasks_env_file.seek(SeekFrom::Start(0))?;
```

Comment:
this is concerning 🤔

do we know for sure that parallel reads of the DB won't result in seeks?